source/base/trilinos_utilities.cc

```diff
@@ -169,6 +169,21 @@ namespace Utilities
     }
   } // namespace Trilinos
 #endif
+
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+  namespace Trilinos
+  {
+    template <class T, class... Args>
+    Teuchos::RCP<T>
+    make_rcp(Args &&...args)
+    {
+      return Teuchos::RCP<T>(new T(std::forward<Args>(args)...));
+    }
```

Comment:
We could avoid a lot of code changes by always using `Utilities::Trilinos::make_rcp` and gaming it an alias for `Teuchos::rcp` in case that is available.
```suggestion
#ifdef DEAL_II_TRILINOS_VERSION_GTE(14, 0, 0)
     using Teuchos::make_rcp;
#else
    template <class T, class... Args>
    Teuchos::RCP<T>
    make_rcp(Args &&...args)
    {
      return Teuchos::RCP<T>(new T(std::forward<Args>(args)...));
    }
#endif
```

Reply:
I like that solution, it is somewhere between the both approaches I suggested.